modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -167,6 +219,58 @@
 					this._processFile(imageFile, event.editor);
 				}
 			}
+			else if (event.data && event.data.type === 'html') {
+				const editor = event.editor;
+				const fragment = CKEDITOR.htmlParser.fragment.fromHtml(
+					event.data.dataValue
+				);
+
+				fragment.forEach((element) => {
+					if (
+						element.type === CKEDITOR.NODE_ELEMENT &&
+						element.name === 'img'
+					) {
```

Comment:
Maybe we could just return early in case this is not an image? That way we wouldn't have to indent as much code.

```js
if (
    element.type !== CKEDITOR.NODE_ELEMENT ||
    element.name !== 'img'
) {
    return;
}

// Do stuff 
```

Reply:
:+1: